python/lsst/obs/hsc/makeCoaddTempExp.py

```diff
@@ -74,26 +77,15 @@ def getCalibratedExposure(self, dataRef, bgSubtracted):
             mi += background.getImage()
             del mi
 
-        if self.config.doApplyUberCal:
-            if self.config.useMeasMosaic:
-                from lsst.meas.mosaic import applyMosaicResultsExposure
-                # NOTE: this changes exposure in-place, updating its Calib and Wcs.
-                # Save the calibration error, as it gets overwritten with zero.
-                calibrationErr = exposure.getPhotoCalib().getCalibrationErr()
-                try:
-                    applyMosaicResultsExposure(dataRef, calexp=exposure)
-                except dafPersist.NoResults as e:
-                    raise MissingExposureError('Mosaic calibration not found: %s ' % str(e)) from e
-                photoCalib = afwImage.PhotoCalib(exposure.getPhotoCalib().getCalibrationMean(),
-                                                 calibrationErr,
-                                                 exposure.getBBox())
-            else:
-                photoCalib = dataRef.get("jointcal_photoCalib")
-                skyWcs = dataRef.get("jointcal_wcs")
-                exposure.setWcs(skyWcs)
+        if self.config.doApplyExternalPhotoCalib:
+            photoCalib = dataRef.get(f"{self.config.externalPhotoCalibName}_photoCalib")
         else:
             photoCalib = exposure.getPhotoCalib()
 
+        if self.config.doApplyExternalSkyWcs:
+            skyWcs = dataRef.get(f"{self.config.externalSkyWcsName}_wcs")
+            exposure.setWcs(skyWcs)
```

Comment:
This is great - the exposure already has a WCS from the calexp (heh) if this doesn't happen, so it makes sense to only setWcs if we are overriding it.
Similarly, isn't it true that the exposure already has a photoCalib from the calexp? If so, then this can be simplified even further by removing the the else above (which retrieves the photoCalib from the exposure if there's no external one) and only calls setPhotoCalib if there is a new one.